Document MQTT lock.open (#20439)
home-assistant/home-assistant.github.io: source/_integrations/lock.mqtt.markdown

Optimistic mode can be forced, even if state topic is available. Try to enable it, if experiencing incorrect lock operation.

> A simple lock only supports commands for lock and unlock. If in addition an open command is needed (e.g. to open the bolt in addition to the latch) a definition of `payload_open` is required in the configuration. In optimistic mode on an `payload_open` command the lock is in the unlocked state.

Reviewer: ```suggestion
It's mandatory for locks to support `lock` and `unlock`. A lock may optionally support `open`, (e.g. to open the bolt in addition to the latch), in this case `payload_open` is required in the configuration. If the lock is i optimistic mode, it will change states to `unlocked` when handling the `open` command.
```

> It's mandatory for locks to support `lock` and `unlock`. A lock may optionally support `open`, (e.g. to open the bolt in addition to the latch), in this case `payload_open` is required in the configuration. If the lock is i optimistic mode, it will change states to `unlocked` when handling the `open` command.

Reviewer 2: ```suggestion
It's mandatory for locks to support `lock` and `unlock`. A lock may optionally support `open`, (e.g. to open the bolt in addition to the latch), in this case, `payload_open` is required in the configuration. If the lock is in optimistic mode, it will change states to `unlocked` when handling the `open` command.
```

> payload_open:

Reviewer 2: This is a lost marker or something?

Reviewer: I guess, removed.

  required: false
  type: string
  default: UNLOCK
payload_open:

>   description: The payload that requesta an open command. Only available if `payload_open` is explicitly specified.

Reviewer: ```suggestion
  description: The payload sent to the lock to `open` it.
```
